py/makeqstrdefs.py

```diff
@@ -39,11 +40,26 @@ def preprocess():
         os.makedirs(os.path.dirname(args.output[0]))
     except OSError:
         pass
-    with open(args.output[0], "w") as out_file:
-        if csources:
-            subprocess.check_call(args.pp + args.cflags + csources, stdout=out_file)
-        if cxxsources:
-            subprocess.check_call(args.pp + args.cxxflags + cxxsources, stdout=out_file)
+
+    def pp(flags):
+        def run(files):
+            return subprocess.check_output(args.pp + flags + files)
+
+        return run
+
+    cpus = multiprocessing.cpu_count()
+    p = multiprocessing.dummy.Pool(cpus)
+    with open(args.output[0], "wb") as out_file:
+        for flags, sources in (
+            (args.cflags, csources),
+            (args.cxxflags, cxxsources),
+        ):
+            if not sources:
+                continue
+            batch_size = (len(sources) + cpus - 1) // cpus
```

Comment:
isn't this like divmod or something else from math module?

Reply:
@stinos it's definitely not builtin `divmod` which is `(a // b, a % b)`. It's the "ceiling integer division". To me this is the idiomatic way to express this and I'm not aware of any other Python standard library or builtin that provides this (although that would be nice if there was such a thing). Open to suggestions!

Reply:
> ceiling integer division

I think you'll find it's floor integer division

Reply:
`math.ceil(len(sources)/cpus) or 1` then ?

Reply:
best not to introduce floats in to this calculation which is conceptually integer based